src/mono/mono/mini/mini-arm64.c

```diff
@@ -5931,6 +5931,15 @@ mono_arch_emit_prolog (MonoCompile *cfg)
 		code = emit_addx_imm (code, cfg->arch.args_reg, ARMREG_FP, cfg->stack_offset);
 	}
 
+	/* Call the init wrapper which checks if the method needs to be initialised or not */
+	/* https://github.com/dotnet/runtime/pull/82711, https://github.com/dotnet/runtime/issues/83378, https://github.com/dotnet/runtime/issues/83379 */
+#ifdef ENABLE_WIP_METHOD_NOLLVM_SELF_INIT
+	if (cfg->compile_aot && !COMPILE_LLVM (cfg)) {
+		code = emit_imm (code, ARMREG_R0, cfg->method_index);
+		code = emit_call (cfg, code, MONO_PATCH_INFO_METHOD, (gconstpointer) mono_marshal_get_aot_init_wrapper (AOT_INIT_METHOD));
```

Comment:
I am not sure if this is a correct way to invoke wrappers generated in: https://github.com/dotnet/runtime/blob/6229f5f403e646f07f94f6c18fff04f04f2dda8e/src/mono/mono/mini/aot-compiler.c#L4470

During the native linking, there is an undefined symbols for architecture arm64:
```
  "_mono_aot_HelloWorld_init_method", referenced from:
      HelloWorld_Program_Main_string__ in mono_aot_0nLeOF.o
      HelloWorld_Program__ctor in mono_aot_0nLeOF.o
      wrapper_other_object_init_method_intptr_intptr in mono_aot_0nLeOF.o
      wrapper_other_object_init_method_gshared_mrgctx_intptr_intptr_intptr in mono_aot_0nLeOF.o
      wrapper_other_object_init_method_gshared_this_intptr_intptr_intptr in mono_aot_0nLeOF.o
      wrapper_other_object_init_method_gshared_vtable_intptr_intptr_intptr in mono_aot_0nLeOF.o
```

I think it comes from method prolog trying to invoke the init wrapper, but I would expect invocation of `wrapper_other_object_init_method_intptr_intptr` instead of `_mono_aot_HelloWorld_init_method`. @vargaz What we are doing wrong?

Reply:
I am not 100% how this should work but I think the body of `_mono_aot_HelloWorld_init_method` is never generated, looking at the compiler output: https://gist.github.com/ivanpovazan/96542ae24088a4a5e735d414b92e4c9e 
there are only jumps to that symbol `bl _mono_aot_HelloWorld_init_method`, but it is not generated anywhere.